src/lib/connector.ts

```diff
@@ -69,13 +49,22 @@ export async function connect(options: ConnectionOptions): Promise<Connection> {
 
   if (
     !providerNetwork?.name ||
-    !SUPPORTED_NETWORKS.find((net) => net.key === providerNetwork.name)
+    !SUPPORTED_NETWORKS.find((net) => {
```

Comment:
Why is supported networks not just a map/object?

Reply:
yeh

Reply:
let's leave for now... but if we find we end up just iterating over it a lot, we might consider doing the object route.